tests/test_oml/test_datasets/test_list_dataest.py

```diff
@@ -0,0 +1,75 @@
+from pathlib import Path
+from typing import Iterator, List, Tuple
+
+import pandas as pd
+import pytest
+import torch
+from torch.utils.data import DataLoader
+
+from oml.const import MOCK_DATASET_PATH
+from oml.datasets.list_ import ListDataset
+
+
+@pytest.fixture
+def images() -> Iterator[List[Path]]:
+    yield list((MOCK_DATASET_PATH / "images").iterdir())
+
+
+def get_images_and_boxes() -> List[Tuple[Path, int, int, int, int]]:
+    df = pd.read_csv(MOCK_DATASET_PATH / "df_with_bboxes.csv")
+    sub_df = df[["path", "x_1", "y_1", "x_2", "y_2"]]
+    sub_df["path"] = sub_df.path.apply(lambda p: MOCK_DATASET_PATH / p)
+    lines: List[Tuple[Path, int, int, int, int]] = [tuple(line[1]) for line in sub_df.iterrows()]  # type: ignore
+    return lines
```

Comment:
To make it more real let's return the structures that we pass to the dataset: list of paths and boxes (`Sequence[Optional[TBBox]]`)

Reply:
see the comment below ->